src/Agent.Plugins/PipelineCache/PipelineCacheServer.cs

```diff
@@ -42,32 +52,150 @@ public class PipelineCacheServer
+                            {
+                                // readStdOut and readStdError should only fail if the process dies
+                                // processTcs.Task cannot fail as we only call SetResult on processTcs
+                                await Task.WhenAll(readStdOut, readStdError, processTcs.Task);
+                            }
+
+                            int exitCode = await processTcs.Task;
+
+                            if (exitCode == 0)
+                            {
+                                context.Output($"Process exit code: {exitCode}");
+                                foreach (string line in output)
+                                {
+                                    context.Output(line);
+                                }
+                            }
+                            else
+                            {
+                                throw new Exception($"Process returned non-zero exit code: {exitCode}");
+                            }
+                        }
+                        catch (Exception e)
+                        {
+                            // Delete archive file.
+                            if(File.Exists(archiveFile))
+                            {
+                                File.Delete(archiveFile);
+                            }
+                            foreach (string line in output)
+                            {
+                                context.Error(line);
+                            }
+                            ExceptionDispatchInfo.Capture(e).Throw();
+                        }
+                    } // end of tarring process.
+
+                    //Upload the pipeline artifact.
+                    PipelineCacheActionRecord uploadRecord = clientTelemetry.CreateRecord<PipelineCacheActionRecord>((level, uri, type) =>
+                        new PipelineCacheActionRecord(level, uri, type, nameof(dedupManifestClient.PublishAsync), context));
+                    PublishResult result = await clientTelemetry.MeasureActionAsync(
+                        record: uploadRecord,
+                        actionAsync: async () =>
+                        {
+                            return await dedupManifestClient.PublishAsync(archiveFile, cancellationToken);
+                        });
+
+                    CreatePipelineCacheArtifactOptions options = new CreatePipelineCacheArtifactOptions
+                    {
+                        Fingerprint = fingerprint,
+                        RootId = result.RootId,
+                        ManifestId = result.ManifestId,
+                        ProofNodes = result.ProofNodes.ToArray(),
+                        ContentFormat = ContentFormatConstants.SingleTar
+                    };
+                    // delete archive file.
+                    if(File.Exists(archiveFile))
+                    {
+                        File.Delete(archiveFile);
+                    }
```

Comment:
needs to be encased into try/catch where catch does nothing:

try
{
File.Delete(archiveFile);
}
catch { }